include/rear-panel/core/messages.hpp

```diff
@@ -135,10 +135,51 @@ struct DeviceInfoResponse : BinaryFormatMessage<MessageType::DEVICE_INFO_RESP> {
     auto operator==(const DeviceInfoResponse& other) const -> bool = default;
 };
 
+struct EnterBootlader : BinaryFormatMessage<MessageType::ENTER_BOOTLOADER> {
+    uint16_t length;
+
+    template <bit_utils::ByteIterator Input, typename Limit>
+    static auto parse(Input body, Limit limit) -> EnterBootlader {
+        uint16_t type = 0;
+        uint16_t len = 0;
+        body = bit_utils::bytes_to_int(body, limit, type);
+        body = bit_utils::bytes_to_int(body, limit, len);
+        if (len > 0) {
+            // we got something wrong from this figure out what we want to
```

Comment:
This actually brings up the point that `binary_parse` doesn't have a way to fail parsing AFAIK. Doesn't have to be in this PR, but I think it'd make sense for the `Parsable` concept to dictate that `parse` returns an _optional_ value so there's a way to propagate a parsing error.

Reply:
upgraded the parsing so that if a body of a message is invalid it will return monostate, and the serial handling task will automatically send out a AckFailed if a message on the serial bus is not parsed into a valid message